Trial branch

isibley765 opened the pull request:
I've created the basic layout and accessibility of the website. Right now the only connection to the database is a check to see if it exists, with a creation if it does not.

Username and password fields accept anything at the moment, for the sake of accessibility until I add more capabilities. Since there's nothing to specify whether the user already has a password or needs to create one, I've just had both fields pop up for now.

The options>help dropdown item on the top right, and the select weekday on the toolbar on the right both work as well. The weekday select also makes a value change on an object within the ajax code that will allow me to pass back information to the database more easily, but that is not apparent to the user.

isibley765 committed:
Creates the basic html page

makes a basic layout of the concept for the website

isibley765 committed:
The Basics

I've created the basic layout and accessibility of the website. Right
now, the only connection to the database that's made is a check to see
if it exists, and a creation of it if it does not.

Username and passwords accept anything. There is no specification yet
for the username choosing which password/create password field pops up,
so both come up presently, and accept anything (for the means of getting
access to the webpage).

The options>help choice on the top right and choosing the day of the
week both work as well. Adding a refresh capability is currently not
present; I will explore adding that in once I've got everything I've
already learned implemented.

isibley765 committed:
Remaining markers for the tag associations I'd created.

Missed a few tag markers in my last commit, wanted to make sure I was as
clear as possible.

bsalter commented on app/mysql/init.js:
```diff
@@ -0,0 +1,22 @@
+      });
+    });
+    
+}
```
This looks the same as what's in connect.js. Why is this repeated?

isibley765 replied:
Originally, I wasn't getting connect.js to work with both creating the connection and checking for/setting up the database. I've deleted that file now, since I got it to work in connect.js, but it looks like I managed to do the deletion after I sent the pull request and didn't catch it. It doesn't affect functionality though, I didn't get far enough that anything depended on it.

bsalter replied:
If you push that change, it should automatically update the pull request.

bsalter commented on app/webpage/format.css:
```diff
@@ -0,0 +1,129 @@
+/* plan-tlbr-03 */
+#tool-buttons{
+  padding-top: 3rem;
+}
```
You use !important quite a lot throughout this CSS. This is considered bad practice - you should use !important sparingly. Instead, make sure your CSS rules are scoped correctly.

isibley765 replied:
I was finding that I wasn't able to change Bootstrap's defaults without it. I'm happy to go back and see if I can find a different way, I only resorted to that after finding nothing else for a good while.

bsalter replied:
You should be able to override bootstrap's defaults.

isibley765 replied:
Did it. I didn't realize being just as specific later wasn't good enough, I was coming across a lot of people that were resorting to this method like I did before I got lucky with someone who knew what he was doing. Should all be gone now, I pushed it through. That, and the copied init.js file is now deleted, it looks like the comments from that left with it as well.

isibley765 replied:
Scratch that, it just hid the comments since they were outdated. Good to know.

bsalter commented on app/webpage/homepage.html:
```diff
@@ -0,0 +1,312 @@
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js" integrity="sha384-0mSbJDEHialfmuBBQP6A4Qrprq5OVfW37PRR3j5ELqxss1yVqOtnepnHVP9aJ7xS" crossorigin="anonymous"></script> 
+    <link rel="stylesheet" href="http://getbootstrap.com.vn/examples/equal-height-columns/equal-height-columns.css" />
+    <link href="format.css" rel="stylesheet" type="text/css" />
+    <script src="main.js"></script>
```
Revision for the future: use bower & gulp or grunt instead of installing these scripts by hand. This is ok for now, we'll discuss later.

bsalter commented on app/webpage/homepage.html:
```diff
@@ -0,0 +1,312 @@
+        
+          <div class="week weekend Sunday">
+            <div class="day">Sunday</div>
+            <div class="task">
```
This is not a good solution. Let's discuss. Is this just a placeholder?

isibley765 replied:
Of sorts. I wanted to get the general layout on the webpage, and this was the best shorthand I knew to do, I was aiming at including jQuery there when I addressed pulling tasks from the database.. So yes. I'll be going in and changing things after I get the capability to pull the user's saved tasks, and I was thinking that jQuery would certainly be less bulky, but I'd have to come up with a way to address scalability.
But I'd be happy to hear better ideas, especially than what's there now. This serves a much better purpose looking at it on the webpage than it does the html code, I'd be more than happy to change it.

bsalter commented:
In the future please keep PRs much smaller than this. PRs should be frequent, and should tackle a particular issue, defect or feature. This PR is very long and that makes it tough to review.

isibley765 committed:
Unnecessary file

isibley765 committed:
Stopped using !important

Surprisingly short search showed me something that I hadn't discovered
before, and the post explained the weights of classes and such, and how
to be more specific than bootstrap. I don't know if it's bad coding or
not to add a class or id to the body tag, it did seem the most ideal for
overriding any part of the page body though.

bsalter commented on app/webpage/main.js:
```diff
@@ -0,0 +1,97 @@
+
+/* lgin-user-01 */
+function eValidate() {
+  if($("#email").val() == "" || $("#email").val() == undefined){
```
Let's put spaces around {

bsalter replied:
Also, you should save the value to a variable if you are going to access it twice, because otherwise jQuery searches the DOM both times.

```
var current_email = $("#email").val();
if(current_email...
```

isibley765 replied:
Question....
I'm going to have to be making an Ajax call later in that function. Would it still be efficient to just save $("#email") to a variable, rather than $("#email").val()? That way I could call the identity to .hide() later too.
I don't know how much searching .val() might do, but I wanted to double check that it would work.

bsalter commented on app/webpage/main.js:
```diff
@@ -0,0 +1,97 @@
+  if($("#email").val() == "" || $("#email").val() == undefined){
+    alert("Please enter your email address.");
+    custTitle();
+  }else{
```
} else {

isibley765 replied:
Should all be fixed. I had some of these later too, so I added space there as well.

isibley765 committed:
Tag list converted, more tweaks

Converted the tag list to a .js file for visibility, and made a few more
changes that Ben pointed out.

isibley765 committed:
Small typo in pswdValidate function

isibley765 committed:
Using dotenv to store database env. var.'s

Basic implimentation of dotenv. There's a few more minor changes, most
notably the dependencies in package.json. I ran across something that
told me that field told what packages my project depended on, so I
figured I'd fill it out. Nothing major.

isibley765 committed:
Revert "Using dotenv to store database env. var.'s"

This reverts commit 6f58ce910ceda2eb5886b873c80cdcf2d7dfd4fd.
